### PR TITLE
Incorporate changes from OBS linter

### LINKED ISSUE #2178: Fix issues shown by OBS linter for major distros
Fix some of the issues shown to exist in SLES style distros when creating service RPMs with our spec file.  These include using the SLES RPM macros for adding units which are different than the RHEL style macros.

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Fixes #2178

### CODE REVIEW

File: service/geopm-service.spec.in [hunk 1]
```diff
@@ -48,15 +48,19 @@ BuildRequires: systemd-devel
 %define python_major_version 3
 BuildRequires: python3
 BuildRequires: python3-devel
+BuildRequires: python3-setuptools
 %define python_bin %{__python3}
 
 %if 0%{?suse_version}
+BuildRequires: libdbus-1-3
```
Comment (dannosliwcd) [inline service/geopm-service.spec.in:60]: I think I was wrong to recommend this in our earlier discussion. There's an OBS linter warning about this:
> geopm-service.x86_64: W: explicit-lib-dependency libdbus-1-3
You must let rpm find the library dependencies by itself. Do not put unneeded
explicit Requires: tags.

Is our actual dependency from geopm-service on `dbus-1`, maybe?

File: service/geopm-service.spec.in [hunk 2]
```diff
@@ -38,59 +41,98 @@ Vendor: Intel Corporation
 URL: https://geopm.github.io
 Source0: geopm-service.tar.gz
 BuildRoot: %{_tmppath}/geopm-service-%{version}-%{release}-root
+Prefix: %{_prefix}
+
 BuildRequires: gcc-c++
 BuildRequires: unzip
 BuildRequires: libtool
 BuildRequires: systemd-devel
```
Comment (dannosliwcd) [inline service/geopm-service.spec.in]: I guess we require version 221 or higher, for the sd-bus interface. If our supported OSs all have at least that version, then I guess the version annotation is not needed in this file. If not, maybe clarify the note in https://geopm.github.io/requires.html#systemd-library-requirement to specify why we require newer distributions for the service.

File: service/geopm-service.spec.in (hunk 1 above)
Comment (cmcantalupo) [reply service/geopm-service.spec.in:60]: I have removed the Requires: section, which is what the linter is complaining about.  I have left the BuildRequires.

**Review by dannosliwcd — APPROVED** [review]
LGTM

**Pull request merged**